src/controllers/genres.controller.ts

```diff
@@ -28,10 +28,20 @@ export class GenresController {
     return genre;
   }
 
-  async deleteGenre(req: Request): Promise<IGenreDocument | null> {
+  async deleteGenre(
+    req: Request,
+    res: Response,
+  ): Promise<IGenreDocument | null | undefined> {
     const { id } = req.params;
-    const movie = await this.genresService.deleteGenre({ id });
-    return movie;
+    const genre = await this.genresService.findById({ id });
+
+    if (!genre) {
+      res.status(404).json({ message: "Genre not found" });
+      return;
+    }
+
+    const deletedGenre = await this.genresService.deleteGenre({ id });
```

Comment:
Maybe this will be better?
`    if (!deletedGenre) {
      return res.status(404).json({ message: `cannot find any genre with ID ${id}` });
    }`

Reply:
Yes, I agree that this is a better option. Updated to: 
<img width="392" alt="image" src="https://github.com/YevheniiaBondarieva/Node.js_Course_Practice/assets/106626022/1b8c8123-f7c4-4869-804a-89f390a98172">